qiskit/result/mitigation/local_readout_mitigator.py

```diff
@@ -253,9 +254,10 @@ def assignment_matrix(self, qubits: List[int] = None) -> np.ndarray:
             qubits = self._qubits
         if isinstance(qubits, int):
             qubits = [qubits]
-        mat = self._assignment_mats[qubits[0]]
-        for i in qubits[1:]:
-            mat = np.kron(self._assignment_mats[qubits[i]], mat)
+        qubit_indices = [self._qubit_index[qubit] for qubit in qubits]
+        mat = self._assignment_mats[qubit_indices[qubits[0]]]
+        for i in qubit_indices[1:]:
+            mat = np.kron(self._assignment_mats[i], mat)
```

Comment:
The two forms of indexing `self._assignment_mats` look inconsistent here. Line 258 is `self._assignment_mats[qubit_indices[qubits[0]]]`, then the first iteration of the loop is `self._assignment_mats[qubit_indices[1]]`   There's an extra level of indirection in the first one.

Can you add a test when the `qubits` input is supplied out-of-order (`qubits=[2, 1, 0]`), to check/catch this?

Reply:
Thanks! It's a real bug (in 258 it should be `mat = self._assignment_mats[qubit_indices[0]]`). I've fixed it and added a test that catches this.